libs/saveFiles.js

```diff
@@ -38,22 +38,29 @@ const sanitizeEntry = function (entry) {
   return entry
 }
 
-const downloadEntry = function (entry, folderPath) {
-  const reqOptions = Object.assign(
-    {
-      uri: entry.fileurl,
-      method: 'GET',
-      jar: true
-    },
-    entry.requestOptions
-  )
+const downloadEntry = function (entry, options) {
+  const reqOptions = Object.assign({
+    uri: entry.fileurl,
+    method: 'GET',
+    jar: true
+  }, entry.requestOptions)
+
   return cozy.files
-    .statByPath(folderPath)
+    .statByPath(options.folderPath)
     .then(folder => {
-      return cozy.files.create(rq(reqOptions), {
+      let filePromise = rq(reqOptions)
+      const createFileOptions = {
         name: getFileName(entry),
-        dirID: folder._id
-      })
+        dirID: folder._id,
+        contentType: options.contentType
+      }
+      if (options.fileStreamProcess) {
+        return filePromise
+        .then(data => options.fileStreamProcess(data))
+        .then(data => cozy.files.create(data, createFileOptions))
```

Comment:
Here instead of returning early and creating a new branch in the code, I think it would be simpler to do

```
if (options.fileStreamProcess) {
  filePromise = filePromise.then(data => options.fileStreamProcess(data))
}
```

to process the file, this way you do not have to repeat cozy.files.create(data, createFileOptions).

Reply:
That is what I did at first, and it works when we use the fileStreamProcessOption. But in the case we do not use it, the result of the request promise is not a stream and cannot be taken by cozy.files.create. Then I add to change the result of the request promise to a stream and I found it much simpler to create a new branch in the code. I suppose I should add some comments.